external/fv3fit/fv3fit/keras/_models/shared/halos.py

```diff
@@ -45,7 +45,8 @@ def _quantities_from_dataset(
     ds: xr.Dataset, n_halo: int
 ) -> Mapping[Hashable, pace.util.Quantity]:
     return_dict = {}
-    nx, ny, nz = ds.dims["x"], ds.dims["y"], ds.dims["z"]
+    nx, ny = ds.dims["x"], ds.dims["y"]
+    nz = ds.dims["z"] if "z" in ds.dims else 0
```

Comment:
Setting this to zero makes me nervous because it will result in size-zero arrays if an array with a z-dimension is requested. No z-dimensions should get requested in this routine, so this would be a way to make sure that's the case and error more loudly if something breaks.

```suggestion
    # if nz is not in the dataset, should not get used
    # we set it to -1 to cause errors if used
    nz = ds.dims["z"] if "z" in ds.dims else -1
```